views.py

```diff
@@ -106,6 +111,11 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
                         response.content = "<h1>" + content_msg + "</h1>"
                     return response
 
+    elif bag_modified == "true" and metadata_dirty == 'true':
+        if path.endswith("resourcemap.xml") or path.endswith('resourcemetadata.xml'):
+            # we need to regenerate the metadata xml files
+            create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+
```

Comment:
I have changed all calls to ```resource_modified``` by passing the optional parameter ```overwrite_bag=False``` which prevents bag creation.